src/webots/vrml/WbNode.cpp

```diff
@@ -1442,8 +1442,23 @@ WbNode *WbNode::createProtoInstance(WbProtoModel *proto, WbTokenizer *tokenizer,
 
   // 2. create the parameters list from the model (default values)
   QVector<WbField *> parameters;
-  foreach (WbFieldModel *model, protoFieldModels)
-    parameters.append(new WbField(model, NULL));
+  QVector<QMap<QString, WbNode *>> parametersDefMap;
+  bool hasDefaultDefNodes = false;
+  foreach (WbFieldModel *model, protoFieldModels) {
```

Comment:
```suggestion
  foreach (const WbFieldModel *model, protoFieldModels) {
```

We saw in the past that Qt `foreach` performance is very bad with non-const items.